### ISSUE TITLE
setting a default `then` results in infinite requests

### ISSUE THREAD

**Issue by nfcampos** [heroku/react-refetch#119]
We didn't realise when we added the capability to set default `then`, `andThen`, etc. that it results in infinite requests. Because the defaults also apply to the `then` request so this
```js
connect.defaults({then: '/other/url' })(() => ({
  abc: '/some/url'
})
```
becomes
```js
connect(() => ({
  abc: {
    url: '/some/url'
    then: {
      url: '/other/url'
      then: {
        url : '/other/url'
        ...
      }
    }
  }
})
```
First, we didn't realise this at the time because 2 tests that run before those for default `then` replace `window.fetch` and don't restore it back to the original mock. As such, the test requests never `resolve` and it seems to work... I'm fixing the tests, I'll open a PR soon. (done: see #120)

Second, I'm not sure how to actually fix this, do we have to special case `then` mappings so that they don't get the default `then`? How about `andThen` mappings, they also don't get the default `andThen`?

**Comment by nfcampos** [comment]
Thinking about this a bit more it seems that default `andThen` and default `andCatch` are fine because the comparison prevents the requests from going forward. (if the default `andThen` has `force: true` it might fall into this bug as well)

**Comment by jsullivan** [comment]
@nfcampos I think you're right about `andThen` and `andCatch`. I need to give this more thought because I haven't come up with a way to fix this that I'm happy with yet either.

**Comment by nfcampos** [comment]
@jsullivan yeah I'm not sure how to fix this bug and still cleanly support setting a default for `then`. I don't see any way of doing so that doesn't treat `then` mappings as somehow special, which I'd like to avoid

**Comment by ryanbrainard** [comment]
What about making it where `then` is only applied if it is different from the parent's `then`? I can't think of a good recursive case, but if there is one, apps could always re-create the function by using `Function.prototype.bind()` (kind of hacky, but maybe ok for such a corner case?). I'll poke around the code to see if this will work...

**Comment by ryanbrainard** [comment]
Fixed by #123
